elastic/kibana
# [APM][RUM] adjust data types for uiFilters and range in APM requests

## Summary

Fixes elastic/uptime-dev#56

This PR adjusts data type for the query in has_rum_data route. 

This is primarily to resolve issues with the hasRumData route, the params for which can be entirely undefined. 

```
const rumHasDataRoute = createApmServerRoute({
  endpoint: 'GET /api/apm/observability_overview/has_rum_data',
  params: t.partial({
    query: t.intersection([uiFiltersRt, rangeRt]),
  }),
  options: { tags: ['access:apm'] },
  handler: async (resources) => {
    const setup = await setupRequest(resources);
    return await hasRumData({ setup });
  },
});
```
In particular, the exploratory view uses this api route to dynamically fetch apm indicies, without passing any params. This caused the error ```Invalid value undefined supplied to : strict_keys/0: Partial<{ query: ({ uiFilters: string } & { start: isoToEpochRt, end: isoToEpochRt }) }>/query: ({ uiFilters: string } & { start: isoToEpochRt, end: isoToEpochRt })/0: { uiFilters: string }/uiFilters: string``` when debug settings were applied. 

**To test**
For APM exploratory view data, it's best to point your local kibana to edge ES.
In the advanced settings, turn Observability debug on
![image](https://user-images.githubusercontent.com/11356435/117066259-86e4b580-acf6-11eb-9be9-be5d3eb580a5.png)
Navigate to exploratory view using the "Analyze Data" button in User Experience or Uptime
Select the data type as "User Experience (RUM)".
Ensure that the two report types "KPI over time" and "Performance Distribution" appear, and that there are no errors from api calls in the network tab. 
<img width="533" alt="Screen Shot 2021-05-04 at 4 36 13 PM" src="https://user-images.githubusercontent.com/11356435/117066552-dc20c700-acf6-11eb-9d36-1e4f63789e13.png">

This API is also used by the Observability Overview page UX section. We should also smoke test that section to make sure it's still working. 

[ ] Observability Overview UX page is still working in both debug and regular mode
[ ] Exploratory View "User Experience (RUM)" data type works in both debug and regular mode
